algorithms/kernel/dbscan/oneapi/dbscan_kernel_ucapi.h

```diff
@@ -0,0 +1,108 @@
+                             daal::data_management::NumericTable * ntAssignments, daal::data_management::NumericTable * ntNClusters,
+                             daal::data_management::NumericTable * ntCoreIndices, daal::data_management::NumericTable * ntCoreObservations,
+                             const Parameter * par);
+
+private:
+    services::Status processResultsToCompute(DAAL_UINT64 resultsToCompute, daal::data_management::NumericTable * ntData,
+                                             daal::data_management::NumericTable * ntCoreIndices,
+                                             daal::data_management::NumericTable * ntCoreObservations);
+    services::Status pushNeighborsToQueue(const oneapi::internal::UniversalBuffer & distances, const oneapi::internal::UniversalBuffer & chunkOffests,
+                                          uint32_t rowId, uint32_t clusterId, uint32_t chunkOffset, uint32_t numberOfChunks, uint32_t nRows,
+                                          uint32_t qEnd, algorithmFPType eps, oneapi::internal::UniversalBuffer & assignments,
+                                          oneapi::internal::UniversalBuffer & queue);
+
+    services::Status countOffsets(const oneapi::internal::UniversalBuffer & counters, uint32_t numberOfChunks,
+                                  oneapi::internal::UniversalBuffer & offsets);
+
+    services::Status setBufferValue(oneapi::internal::UniversalBuffer & buffer, uint32_t index, int value);
+
+    services::Status setBufferValueByQueueIndex(oneapi::internal::UniversalBuffer & buffer, const oneapi::internal::UniversalBuffer & queue,
+                                                uint32_t posInQueue, int value);
+
+    services::Status getPointDistances(const oneapi::internal::UniversalBuffer & data, uint32_t nRows, uint32_t rowId, uint32_t dim,
+                                       uint32_t minkowskiPower, oneapi::internal::UniversalBuffer & pointDistances);
+
+    services::Status getQueueBlockDistances(const oneapi::internal::UniversalBuffer & data, uint32_t nRows,
+                                            const oneapi::internal::UniversalBuffer & queue, uint32_t queueBegin, uint32_t queueBlockSize,
+                                            uint32_t dim, uint32_t minkowskiPower, oneapi::internal::UniversalBuffer & queueBlockDistances);
+
+    services::Status countPointNeighbors(const oneapi::internal::UniversalBuffer & assignments,
+                                         const oneapi::internal::UniversalBuffer & pointDistances, uint32_t rowId, int chunkOffset, uint32_t nRows,
+                                         uint32_t numberOfChunks, algorithmFPType epsP, const oneapi::internal::UniversalBuffer & queue,
+                                         oneapi::internal::UniversalBuffer & countersTotal, oneapi::internal::UniversalBuffer & countersNewNeighbors);
```

Comment:
Each of these functions has many input parameters. Can we eliminate any using private fields of this class?

Reply:
I tried it by realized that resulting data flow is very unclear when part of important parameters is hidden from user.